mdp/nodes/recursive_expansion_nodes.py

```diff
@@ -0,0 +1,767 @@
+        recursion formula. E.g. the first order function value.
+    :type special: int
+
+    :param n: The order of the function to be computed at this step.
+    :type n: int
+
+    :param cur_var: The index of the current variable to be considered in the
+        recursion. This value will have to be lower than x.shape[1].
+    :type cur_var: int
+
+    :param pos: The index of the element to be computed, along the second
+        dimension of result.
+    :type int: int
+
+    :returns: The vectorized result (along the observations) of the
+        n-th recursion step of the cur_var-th variable.
+    :rtype: numpy.ndarray
+    """
+    return result[:, pos-1]*result[:, special]
+
+
+def init_legendre_poly(result, x, pos, cur_var):
+    """Initialize the first and second order before starting the recursion.
+
+    Mathematically this amounts to
+    P_1 = x
+    P_2 = 3/2 * P_1*P_1 - 1/2 .
+
+    .. note::
+
+        The procedure on how an init function is built can be found in the
+        docstring of the init function for
+        standard polynomials *init_standard_poly*.
+
+
+    :param result: Contains the observations along the first dimension
+        and the function values of expansion w.r.t. an observation
+        along the second dimension.
+    :type result: numpy.ndarray
+
+    :param x: The data to be expanded.
+    :type x: numpy.ndarray
+
+    :param pos: The index of the element to be computed, along the second
+        dimension of result.
+    :type int: int
+
+    :param cur_var: The index of the current variable to be considered in the
+        recursion. This value will have to be lower than x.shape[1].
+    :type cur_var: int
+
+    :returns: The index of the next element that will be computed in the
+        recursion, the order of the next element, and the index of a special
+        member.
+    :rtype: Tuple[int, int, int]
+    """
+    result[:, pos] = x[:, cur_var]
+    # after the first variable this is needed (no zero order available)
+    result[:, pos+1] = 3./2.*result[:, pos] * result[:, pos] - .5
+    return pos+2, 3, pos
+
+
+def recf_legendre_poly(result, x, special, n, cur_var, pos):
+    """Implementation of the recursion formula for Legendre polynomials.
+    The recursion formula is Bonnet's recursion formula
+    P_n = (2n-1)/n * x * P_ {n-1}  - (n-1)* P_{n-2}.
+
+        .. note::
+
+        The procedure on how an recursion function is built can be found in the
+        docstring of the recursion function for
+        standard polynomials *recf_standard_poly*.
+
+    :param result: Contains the observations along the first dimension
+        and the function values of expansion w.r.t. an observation
+        along the second dimension.
+    :type result: numpy.ndarray
+
+    :param x: The data to be expanded.
+    :type x: numpy.ndarray
+
+    :param special: Index of a special element to be considered in the
+        recursion formula. E.g. the first order function value.
```

Comment:
Please replace the generic explanation here and in all other recf functions except the standard one. Describe the specific role of ``special`` for Legendre polynomials etc...

Reply:
Done.